pulsar-functions/runtime/src/main/java/org/apache/pulsar/functions/worker/WorkerConfig.java

```diff
@@ -273,6 +273,11 @@
         doc = "The frequency of instance liveness check, in milliseconds"
     )
     private long instanceLivenessCheckFreqMs;
+    @FieldContext(
+            category = CATEGORY_CLIENT_SECURITY,
+            doc = "Whether to enable the broker client authentication used by function workers to talk to brokers"
+    )
+    private boolean brokerClientAuthenticationEnabled = false;
```

Comment:
Don't use `boolean`. This would cause a compatibility issue. The worker service is not able to start with an old configuration.

Use boxed `Boolean`.

```
private Boolean brokerClientAuthenticationEnabled = null;
```

Then add an implementation `isBrokerClientAuthenticationEnabled` to check if `brokerClientAuthenticationEnabled` is set or not. If `brokerClientAuthenticationEnabled` is not set, fall back to use `authenticationEnabled`.

```
public boolean isBrokerClientAuthenticationEnabled() {
    if (brokerClientAuthenticationEnabled != null) {
        return brokerClientAuthenticationEnabled;
    } else {
        return authenticationEnabled;
    }
}
```

Reply:
@sijie Done.